src/Microsoft.Extensions.Logging.Console/ConsoleLoggerProvider.cs

```diff
@@ -45,20 +45,32 @@ public ConsoleLoggerProvider(IConsoleLoggerSettings settings)
 
         private void OnConfigurationReload(object state)
         {
-            // The settings object needs to change here, because the old one is probably holding on
-            // to an old change token.
-            _settings = _settings.Reload();
+            try
+            {
+                // The settings object needs to change here, because the old one is probably holding on
+                // to an old change token.
+                _settings = _settings.Reload();
 
-            foreach (var logger in _loggers.Values)
+                foreach (var logger in _loggers.Values)
+                {
+                    logger.Filter = GetFilter(logger.Name, _settings);
```

Comment:
This would not handle the following suggested by @glennc as you would have already modified state before hitting an exception

> I think the ideal experience here would be that if any errors occur during reloading of configuration we leave everything as it was before the config was changed, so none of your changes apply until you fix whatever is breaking and then all of them take affect. The reason I think that is because we should also log the fact that an error happened and it seems most likely to be log successfully if we ignore all the changes that caused an exception.

Our default implementation ([ConfigurationConsoleLoggerSettings](https://github.com/aspnet/Logging/blob/dev/src/Microsoft.Extensions.Logging.Console/ConfigurationConsoleLoggerSettings.cs)) always seems to read from settings file without caching them because of which you would loose the earlier state of good settings.